src/main/scala/actors/UnattachedAtomActor.scala

```diff
@@ -0,0 +1,104 @@
+package actors
+
+import akka.actor.{Actor, Props}
+import akka.event.{DiagnosticLoggingAdapter, Logging}
+import akka.stream.ActorMaterializer
+import com.typesafe.config.Config
+import actors.messages._
+import com.amazonaws.auth.{AWSCredentialsProviderChain, EnvironmentVariableCredentialsProvider, InstanceProfileCredentialsProvider}
+import com.amazonaws.auth.profile.ProfileCredentialsProvider
+import com.amazonaws.services.dynamodbv2.{AmazonDynamoDB, AmazonDynamoDBClientBuilder}
+import com.gu.contentatom.thrift.{ChangeRecord, User}
+import com.gu.scanamo._
+import com.gu.scanamo.syntax._
+import models.UnattachedAtom
+import vidispine.UpdateXmlGenerator
+
+import scala.concurrent.Future
+import scala.util.{Failure, Success}
+import java.time.LocalDateTime
+
+class UnattachedAtomActor(config:Config) extends Actor {
+  import context.dispatcher
+  implicit val materializer:ActorMaterializer = ActorMaterializer()
+
+  private val maxRetries = 10
+  private val tableName = config.getString("unattached_atoms_table")
+  private val region = config.getString("region")
+
+  protected implicit val logger:DiagnosticLoggingAdapter = Logging.getLogger(this)
+
+  override def receive: Receive = {
+    case MasterNotFound(atomId,createdChange,updatedChange, attempt)=>
+      val origSender = sender() //we message the sender back in the onComplete handler, i.e. a different thread.
+                                //so, we must save a copy of the sender ref here as the subthread doesn't have one.
+
+      logAtomUnattached(atomId, createdChange, updatedChange).onComplete({
+        case Success(maybeResult)=>maybeResult match {
+          case None=> //we expect no data back from an insert
+            logger.info("Saved unattached atom to table")
+            origSender ! SuccessfulSend
+          case Some(response)=>response match {
+            case Left(error)=>
+              logger.error(s"Unable to record missing item: $error")
+              if(attempt>=maxRetries){
+                logger.error(s"Not able to record after $maxRetries attempts, giving up.")
+                sender ! ErrorSend(s"Not able to record after $maxRetries attempts, giving up.")
+              } else {
+                //retry after a short delay
+                Thread.sleep(2000)
+                self ! MasterNotFound(atomId, createdChange, updatedChange, attempt + 1)
```

Comment:
If this retry logic is executed I think the origSender will not be notified

Reply:
hmmm, that's right. I don't think that's a big problem as origSender is only really used for testing, to ensure that it processes the messages correctly, but it's not very neat I suppose

Reply:
fixed it in next commit